core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -95,15 +92,26 @@ private int getNewTaskCount()
             return 1;
         }
 
+        // Do not scale tasks until all existing writer tasks are initialized with maxWriterCount
+        if (writerTasksProvider.get().stream().map(TaskStatus::getMaxWriterCount).anyMatch(Optional::isEmpty)) {
+            return 0;
+        }
```

Comment:
should we also check that `scheduledNodes.size() == writerTasksProvider.get().size` to avoid situation where task status info lags behind scaling up?